test/e2e/publish_confirm.test.ts

```diff
@@ -0,0 +1,107 @@
+import { expect } from "chai"
+import { randomUUID } from "crypto"
+import { connect, Connection } from "../../src"
+import { Rabbit } from "../support/rabbit"
+import { eventually } from "../support/util"
+
+describe("publish a message and get confirmation", () => {
+  const rabbit = new Rabbit()
+  let connection: Connection
+
+  beforeEach(async () => {
+    connection = await connect({
+      hostname: "localhost",
+      port: 5552,
+      username: "rabbit",
+      password: "rabbit",
+      vhost: "/",
+      frameMax: 0, // not used
+      heartbeat: 0, // not used
+    })
+  })
+  afterEach(() => connection.close())
+  afterEach(() => rabbit.closeAllConnections())
+
+  it("after the server replies with a confirm, the confirm callback is invoked", async () => {
+    const stream = `my-stream-${randomUUID()}`
+    await rabbit.createStream(stream)
+    let confirmed = false
+    const publisher = await connection.declarePublisher({
+      stream,
+      publisherRef: "my publisher",
+      confirmCb: () => {
+        confirmed = true
+      },
+    })
+
+    await publisher.send(1n, Buffer.from(`test${randomUUID()}`))
+
+    await eventually(async () => {
+      expect((await rabbit.getQueueInfo(stream)).messages).eql(1)
+    }, 10000)
+    expect(confirmed).true
+  }).timeout(10000)
+
+  it("after the server replies with a confirm, the confirm callback is invoked with the publishingId as an argument", async () => {
+    const stream = `my-stream-${randomUUID()}`
+    await rabbit.createStream(stream)
+    let publishingIds: bigint[] = []
+    const publisher = await connection.declarePublisher({
+      stream,
+      publisherRef: "my publisher",
+      confirmCb: (pubId: bigint[]) => {
```

Comment:
Are you sure it's an integer?

Reply:
Yeah, that might be a mistake. When I wrote the tests I still had a bit of confusion between publishERId and publishINGIds

Reply:
Actually, I've now rechecked my code. This looks fine (besides the fact that the cb is in the constructor). Could you elaborate? What did I declare as an integer which shouldn't be? The publishingIds should be, from the specifics, a list of uint64
```PublishConfirm => Key Version PublishingIds
  Key => uint16 // 0x0003
  Version => uint16
  PublisherId => uint8
  PublishingIds => [uint64] // to correlate with the messages sent```
Am I mistaken?